pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -121,6 +129,38 @@ func (f *FluxAddonClient) InstallGitOps(ctx context.Context, cluster *types.Clus
 		machineConfigs:   machineConfigs,
 	}
 
+	if clusterSpec.FluxConfig.Spec.Github != nil {
+		err := f.installGitOpsGithub(ctx, cluster, fc, clusterSpec)
+		if err != nil {
+			return fmt.Errorf("installing GitHub gitops: %v", err)
+		}
+	}
+
+	if clusterSpec.FluxConfig.Spec.Git != nil {
+		err := f.installGitOpsGenericGit(ctx, cluster, fc, clusterSpec)
```

Comment:
Do we need to be explicit about Generic git? Git is not enough? I might like GithubProvider and GitProvider naming better if that is the issue, but that also shows that having a provider interface might be better.

Reply:
to recap discussion with @vivek-koppuru -- 
we're going to take another look at the abstractions we're using here, and ideally implement some more narrow interfaces that we can use for Repository Management and other provider-specific actions, and get this logic out of the Flux workflow. That is going to be done as an ongoing effort after this pr is merged and we have end-to-end tests in place, ideally for the next+1 milestone, allowing us to both release this working feature for users and continuing to improve the overall maintainability. See: https://github.com/aws/eks-anywhere/issues/1993